### PR TITLE
Ftr: delete session.handleLoop

### PR DESCRIPTION
**Pull request by AlexStocks**
**What this PR does**:

delete session.handleLoop to decrease a goroutine of tcp session

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -555,9 +530,7 @@ func (s *session) handlePackage() {
 			rBuf = rBuf[:runtime.Stack(rBuf, false)]
 			log.Errorf("[session.handlePackage] panic session %s: err=%s\n%s", s.sessionToken(), r, rBuf)
 		}
-
-		close(s.rDone)
-		grNum := atomic.AddInt32(&(s.grNum), -1)
+		grNum := s.grNum.Add(1)
```
Comment (watermelo) [inline session.go:533]: `grNum := s.grNum.Add(1)` -> `grNum := s.grNum.Add(-1)`

File: session.go [hunk 2]
```diff
@@ -479,56 +498,12 @@ func (s *session) run() {
...
-			if wsFlag {
-				err := wsConn.writePing()
-				if err != nil {
-					log.Warnf("wsConn.writePing() = error:%+v", perrors.WithStack(err))
-				}
-			}
-			s.listener.OnCron(s)
-		}
+	s.grNum.Add(1)
+	if _, err := defaultTimerWheel.AddTimer(sessionTimerLoop, gxtime.TimerLoop, s.period, s); err != nil {
```
Comment (watermelo) [inline session.go:502]: 内存泄露，如果 session 关闭，该 session 的 `sessionTimerLoop` 还会一直被执行下去。需要清理资源

File: client.go [hunk 3]
```diff
@@ -170,7 +171,7 @@ func (c *client) dialTCP() Session {
 		}
 
 		log.Infof("net.DialTimeout(addr:%s, timeout:%v) = error:%+v", c.addr, connectTimeout, perrors.WithStack(err))
-		<-wheel.After(connectInterval)
+		<-gxtime.After(connectInterval)
```
Comment (watermelo) [inline client.go:174]: `gxtime.After` 异常阻塞，影响重连，这个场景使用 `time.After` 可便于维护，1.14 对 time 有优化。

File: session.go [hunk 4]
```diff
@@ -462,6 +453,34 @@ func (s *session) WriteBytesArray(pkgs ...[]byte) (int, error) {
 	return wlg, nil
 }
 
+func sessionTimerLoop(_ gxtime.TimerID, _ time.Time, arg interface{}) error {
```
Comment (wenxuwan) [inline session.go:456]: change the name to heartbeat?  the function name has no meaning

Comment (AlexStocks) [reply session.go:456]: @wenxuwan agree with u. @georgehao pls change this func name.

File: benchmark/client/main.go [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+				atomic.AddUint64(&trans, 1)
+
+				t := time.Now().UnixNano()
+				msg := buildSendMsg()
+				_, _, err := tmpSession.WritePkg(msg, WritePkgTimeout)
+				if err != nil {
+					log.Printf("Err:session.WritePkg(session{%s}, error{%v}", tmpSession.Stat(), err)
+				}
+
+				atomic.AddUint64(&transOK, 1)
```
Comment (pantianying) [inline benchmark/client/main.go:138]: atomic.AddUint64(&transOK, 1) should on err==nil

**Review by pantianying — APPROVED** [review]

**Comment by wongoo** [comment]
@georgehao  pls fix the conflict

File: session.go [hunk 6]
```diff
@@ -477,56 +495,12 @@ func (s *session) run() {
...
-		case <-wheel.After(s.period):
-			if wsFlag {
-				err := wsConn.writePing()
-				if err != nil {
-					log.Warnf("wsConn.writePing() = error:%+v", perrors.WithStack(err))
-				}
-			}
-			s.listener.OnCron(s)
-		}
+	s.grNum.Add(1)
```
Comment (wongoo) [inline session.go:498]: move the statement `s.grNum.Add(1)` just before  `go s.handlePackage()`

**Comment by wongoo** [comment]
this pr include the following changes:
1. replace `sync/atomic` using `go.uber.org/atomic`
2. upgrade `gxtime.Wheel` to `gxtime.TimerWheel`
3. delete session heartbeat loop handle, and add it into the `TimerWheel` instead to reduce the count of goroutines.
4. add benchmark code

**Comment by AlexStocks** [comment]
> this pr include the following changes:
> 
> 1. replace `sync/atomic` using `go.uber.org/atomic`
> 2. upgrade `gxtime.Wheel` to `gxtime.TimerWheel`
> 3. delete session heartbeat loop handle, and add it into the `TimerWheel` instead to reduce the count of goroutines.
> 4. add benchmark code

Can not agree with u on suggestion 2.

**Review by wenxuwan — APPROVED** [review]

**Review by wongoo — APPROVED** [review]

**Review by georgehao — APPROVED** [review]

**Pull request merged**